scripts/quests/windurst/To_Catch_A_Falling_Star.lua

```diff
@@ -0,0 +1,156 @@
+                [196] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- Supporting Character dialogue after Accepted
+    {
+        check  = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tohopka'] = quest:progressEvent(198, 0, xi.items.STARFALL_TEAR, xi.items.HANDFUL_OF_PUGIL_SCALES)
+        }
+    },
+
+    -- Twinkle Tree in West Saru.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WEST_SARUTABARUTA] =
+        {
+            ['Twinkle_Tree'] =
+            {
+                onTrigger = function(player, npc)
+                    if VanadielHour() <= 3 then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        return
+                    end
+                    player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if VanadielHour() > 3 then
+                        player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                        return
+                    end
+
+                    if npcUtil.tradeHas(trade, xi.items.HANDFUL_OF_PUGIL_SCALES) then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        if npcUtil.giveItem(player, xi.items.STARFALL_TEAR) then
+                            player:confirmTrade()
+                            if(quest:getVar(player, "Prog") == 0) then
+                                quest:setVar(player, "Prog", 1)
+                            end
+                        end
+                    end
+                end
+            }
+        }
+    },
+
+    -- Quest Complete
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Sigismund'] =
+            {
+                onTrigger = function(player, npc)
+                    player:startEvent(197, 0, xi.items.STARFALL_TEAR)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.items.STARFALL_TEAR, 1 } }) then
+                        player:startEvent(199)
+                    end
+                end
+            },
+
+            onEventFinish =
+            {
+                [199] = function(player, csid, option, npc)
+                    player:tradeComplete()
+                    quest:complete(player)
+                    quest:setVar(player, "Prog", 2)
```

Comment:
Because there's an item reward for this quest, let's reform this like:
```suggestion
                    if quest:complete(player) then
                        player:tradeComplete()
                        quest:setVar(player, "Prog", 2)
                    end
```

Reply:
By doing this the quest won't complete unless the player has room in their inventory.